ehr/api-src/org/labkey/api/ehr/SharedEHRUpgradeCode.java

```diff
@@ -120,6 +125,42 @@ else if (methodName.startsWith(IMPORT_FROM_TSV_PREFIX))
 
             _tsvImports.add(new TsvImport(schemaName, queryName, tsvPath));
         }
+        else if (methodName.startsWith(IMPORT_DOMAIN_TEMPLATE))
+        {
+            String[] tsvArguments = methodName.split(";");
+            if (tsvArguments.length != 3)
+            {
+                throw new UnsupportedOperationException("Expected three arguments for importTemplate but got " + (tsvArguments.length - 1));
+            }
+
+            String moduleName = tsvArguments[1];
+            String domainGroup = tsvArguments[2];
+
+            Module module = ModuleLoader.getInstance().getModule(moduleName);
+            if (module == null)
+                throw new NotFoundException("Module '" + moduleName + "' for domain template import not found");
+
+            DomainTemplateGroup templateGroup = DomainTemplateGroup.get(module, domainGroup);
+            if (templateGroup != null)
+            {
+                if (templateGroup.hasErrors())
+                {
+                    throw new UnsupportedOperationException("Domain template group '" + domainGroup + "' has errors: " + StringUtils.join(templateGroup.getErrors(), "\n"));
+                }
+                try
+                {
+                    templateGroup.createAndImport(EHRService.get().getEHRStudyContainer(ContainerManager.getRoot()), EHRService.get().getEHRUser(ContainerManager.getRoot()), true, false);
```

Comment:
I believe this will fail on when bootstrapping or any other time there's no configured EHR container or user. Building a list of templates to process and then doing it during moduleStartupComplete() should let you piggyback on code that does the checks

Reply:
Yep, tests caught that too.  Fixed now.  Skips this step if EHR study container or EHR admin user are not set.